### DISCUSSION TITLE
WiFi Remote background theory?

### DISCUSSION THREAD

**Discussion by WesWesthaver** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#379] [General]
Is there any documentation that describes the protocol/messaging used by the SaluSpa unit's control panel? How was it reverse engineered? I would very much like to know the process that led to the development of this WiFi remote system.

I recently purchased a SaluSpa Paris and would  like to add WiFi control to it but before I do I would like to have a better understanding of the SaluSpa's control interface that this project ties into.

Thanks,
-Wes

**Comment by WesWesthaver** [comment]
OK. So maybe there isn't any documentation but can someone give a general explanation of how this project communicates with the SaluSpa unit? Is it a serial interface? SPI? I2C? What??? I'm just curious how the author of this software approached this task from scratch without knowing how the SaluSpa interface worked.

**Reply by visualapproach** [reply]
Hi! A good reply would take more time than I have for the moment. I'll get back to it later if you want. Meanwhile checkout bwc_docs.xls for some notes.
Thx

**Reply by visualapproach** [reply]
Basically, I examined the wiring between the display and pump with a logic analyser. I recognized the signal patterns from earlier experience as SPI-ish but with a common data line. the audio line wasn't obvious at first, but further tests revealed that. So now I had to figure out clock phase and bit order and stuff, which I did incorrectly at first, but it didn't matter too much. I only saw jibberish as data and after some thinking i checked if the bits was driving the 7-segment display segments. It took a couple of tries to match the bits to the LED-segments, but then it all became clear and I felt super excited. There is also some other data packets that I needed to understand, like the display sending button presses. It took some time... First module was an arduino Nano. Then I hooked up a ESP-01 to the arduino. And after that it's been a few years of development and learning about MQTT, WIFI, ESP8266, PCB design, Javascript, HTML and more. And the people in this forum has helped with different things.

Prototype:
![IMG_3762](https://user-images.githubusercontent.com/22987250/194145782-e9a0184c-edf4-49b6-b21f-75a2deceb062.jpg)
